File: Projeto_6/src/models/Professor.py
from src.models.UsuarioAutenticavel import UsuarioAutenticavel
from src.models.Relatorio import Relatorio
from src.models.TipoUsuario import TipoUsuario


class Professor(UsuarioAutenticavel, Relatorio):
    """
    Classe Professor que herda de UsuarioAutenticavel e implementa a interface Relatorio.
    Representa um professor com área de atuação e disciplinas ministradas.
    """

    def __init__(self, nome='indefinido', email='indefinido', tipo=TipoUsuario.PROFESSOR, senha='indefinido', areaDeAtuacao='indefinida', disciplinasMinistradas=None):
        """
        Inicializa os atributos do professor.

        Args:
            nome (str): Nome do professor.
            email (str): Email do professor.
            tipo (str): Tipo de usuário, por padrão 'Professor'.
            senha (str): Senha do professor para autenticação.
            areaDeAtuacao (str): Área de atuação do professor.
            disciplinasMinistradas (list[str]): Lista de disciplinas ministradas pelo professor.
        """
        super().__init__(nome, email, tipo, senha)
        self._areaDeAtuacao = areaDeAtuacao
        self._disciplinasMinistradas = disciplinasMinistradas or []

    def gerarRelatorio(self):
        """
        Gera um relatório com as informações do professor.

        Returns:
            str: Relatório formatado contendo nome, área de atuação e disciplinas ministradas.
        """
        disciplinas = ", ".join(self._disciplinasMinistradas)
        return (
            f"Nome: {self._nome}\n"
            f"Área de Atuação: {self._areaDeAtuacao}\n"
            f"Disciplinas Ministradas: {disciplinas if disciplinas else 'Nenhuma disciplina cadastrada'}"
        )

    def autenticar(self, senha):
        """
        Realiza a autenticação do professor com base na senha.

        Args:
            senha (str): Senha a ser verificada.

        Returns:
            bool: True se a senha estiver correta, False caso contrário.
        """
        return self._senha == senha

File: Projeto_6/Projeto6_Erlano/src/Professor/Professor.cpp
#include "Professor.h"
#include <iostream>
using namespace std;

Professor::Professor() : UsuarioAutenticavel(), areaDeAtuacao(""), disciplinasMinistradas({}), tipoUsuario(TipoUsuario::PROFESSOR) {}

Professor::Professor(string nome, string email, string tipo, string senha, string areaDeAtuacao, vector<string> disciplinasMinistradas)
    : UsuarioAutenticavel(nome, email, tipo, senha), areaDeAtuacao(areaDeAtuacao), disciplinasMinistradas(disciplinasMinistradas), tipoUsuario(TipoUsuario::PROFESSOR) {}

void Professor::gerarRelatorio() {
    cout << "=== RELATORIO DE PROFESSOR ===" << endl;
    cout << "Nome: " << nome << ", Email: " << email << ", Tipo: " << tipo << endl;
    cout << "Área: " << areaDeAtuacao << ", Disciplinas: ";
    for (auto d : disciplinasMinistradas) cout << d << " ";
    cout << endl << endl;
}

bool Professor::autenticar(string senhaTentativa) {
    return senhaTentativa == senha;
}